sdk/storage/azure-storage-blob/azure/storage/blob/_models.py

```diff
@@ -354,40 +360,40 @@ class ContainerProperties(DictMixin):
 
     Returned ``ContainerProperties`` instances expose these values through a
     dictionary interface, for example: ``container_props["last_modified"]``.
-    Additionally, the container name is available as ``container_props["name"]``.
-
-    :ivar str name:
-        Name of the container.
-    :ivar ~datetime.datetime last_modified:
-        A datetime object representing the last time the container was modified.
-    :ivar str etag:
-        The ETag contains a value that you can use to perform operations
-        conditionally.
-    :ivar ~azure.storage.blob.LeaseProperties lease:
-        Stores all the lease information for the container.
-    :ivar str public_access: Specifies whether data in the container may be accessed
-        publicly and the level of access.
-    :ivar bool has_immutability_policy:
-        Represents whether the container has an immutability policy.
-    :ivar bool has_legal_hold:
-        Represents whether the container has a legal hold.
-    :ivar bool immutable_storage_with_versioning_enabled:
-        Represents whether immutable storage with versioning enabled on the container.
+    Additionally, the container name is available as ``container_props["name"]``."""
+
+    name: Optional[str]
```

Comment:
Hmm, I think this is another case where we need to see which of these are always populated and maybe mark them as non-optional since this is a class that is probably used a bunch. Its unfortunate the pattern we have for these models.